src/actions/profile-view.js

```diff
@@ -609,10 +615,18 @@ function selectRangeOfTracks(
   return (dispatch, getState) => {
     const lastNonShiftClickInformation = getLastNonShiftClick(getState());
 
-    const lastClickedTrack =
+    let lastClickedTrack =
       lastNonShiftClickInformation && lastNonShiftClickInformation.clickedTrack;
     if (!lastClickedTrack) {
-      // TODO get the track reference from the selected thread
+      const selectedThreadIndexes = getSelectedThreadIndexes(getState());
+      const threadIndex = getFirstItemFromSet(selectedThreadIndexes);
+      if (threadIndex !== undefined) {
+        lastClickedTrack = getTrackReferenceFromThreadIndex(
```

Comment:
It's a bit unfortunate that we had to do a lot of track reference <-> thread index conversions. But it's not possible to fix it in this PR.

Reply:
yeah, it's not obvious how we could change this too...